### ISSUE TITLE
Clarification on Mid-Training and Post-Training Integration

### ISSUE THREAD

**Issue by yihangyao** [hkust-nlp/GUIMid#1]
Dear authors,

Thank you for the great work and for releasing the dataset to the community! I found your paper very insightful, especially the introduction of a mid-training stage before post-training in the context of GUI agent domains.

I have a couple of questions regarding the implementation details of the mid-training stage, and I would greatly appreciate your guidance:

(1) In Section 3.3, you mention that “both stages are integrated under a single optimizer and learning rate schedule.” Does this imply that, for optimal performance, mid-training and post-training should be conducted jointly under a unified schedule? Would it be suboptimal to separate the two stages: first completing mid-training, then post-training independently on the GUI data? If so, we cannot obtain good ckpts from mid-training stages, but only ckpts from (mid+post)-training stages?

(2) For the results presented in Table 3 and Figure 2, could you kindly clarify whether the reported checkpoints were trained on the combined (mid-training + post-training) data or only on the mid-training set?

Thank you very much for your time and consideration. I look forward to your response.

**Comment by junleiz** [comment]
Thanks for your questions!

(1) Thank you for your insightful questions! We will include additional ablation studies on the training strategy in the next version.

**Actually you can get good results by conducting post-training from a mid-training ckpt**

 As for Figure 3, except for the last point, all results are obtained via post-training from mid-training checkpoints. For instance, the 150k result in Figure 3, we conduct post-training from the checkpoint trained on approximately 150k examples during the mid-training phase. Note that we do not load the optimizer state—only the model checkpoint is used, which is a more easy-to-use setting. The initial learning rate for post-training is set to match the learning rate when the model is trained on 150K mid-training data.

(2) We apologize for the confusion and will revise the wording in the next version for clarity. Regarding Table 3, all checkpoints are trained on the combined dataset, which includes both mid-training and post-training data. Otherwise the model wiil not generate actions in the correct format.

**Comment by junleiz** [comment]
You can take the scripts of llama_factory as an example how we conduct post-training from a mid-training ckpt
`### model
model_name_or_path: xxx/guimid_checkpoint-6552/

### method
stage: sft
do_train: true
finetuning_type: full
freeze_vision_tower: true  # choices: [true, false]
freeze_multi_modal_projector: true  # choices: [true, false]
train_mm_proj_only: false  # choices: [true, false]
deepspeed: examples/deepspeed/ds_z2_config_text.json

### dataset
dataset: phone_web_traj
template: qwen2_vl
cutoff_len: 8192
max_samples: 1000000000
overwrite_cache: False
preprocessing_num_workers: 256
dataset_dir: /zhangjunlei/dataset/new_dataset


### output
output_dir: xxxt/guimid_checkpoint-6552_sft
logging_steps: 10
save_strategy: epoch
plot_loss: true
overwrite_output_dir: true
save_total_limit: 1

### train
per_device_train_batch_size: 2
gradient_accumulation_steps: 2
learning_rate: 1.2529740669042581e-05
num_train_epochs: 1.0
lr_scheduler_type: cosine
warmup_ratio: 0.0
bf16: true
ddp_timeout: 180000000
image_max_pixels: 1048576
report_to: wandb
mix_strategy: concat
use_fast_tokenizer: true
disable_shuffling: true
`

**Comment by yihangyao** [comment]
Hi Junlei,

Thanks for your detailed responses, they are very helpful! I'm looking forward to seeing more excellent work from you and your team in the future. 

Thanks again for your time and attention!

Best wishes,

Yihang
